File: tmf.web/Hubs/ChatHub.js
$(function () {

    
    // Proxy created on the fly
    var chat = $.connection.chat;

    // Declare a function on the chat hub so the server can invoke it
    chat.addMessage = function (message) {
        $('#messages').append('<li>' + message + '</li>');
    };

    $("#broadcast").click(function () {

        
        // Call the chat method on the server
        chat.send($('#msg').val())
              .done(function () {
                  console.log('Success!');
              })
              .fail(function (e) {
                  console.warn(e);
              });
    });

    // Start the connection
    $.connection.hub.start();
});

File: tmf.web/Hubs/ChatHub.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using SignalR.Hubs;

namespace tmf.web.Hubs
{
    public class Chat : Hub
    {
        private static ConcurrentDictionary<string, string> _chat = new ConcurrentDictionary<string, string>(StringComparer.OrdinalIgnoreCase);

        public void Send(string message)
        {
            _chat.TryAdd(message, "test");

            // Call the addMessage method on all clients
            Clients.addMessage(message);
        }
    }
}